sql/structured.go

```diff
@@ -138,6 +138,18 @@ func (desc *IndexDescriptor) containsColumnID(colID ColumnID) bool {
 	return false
 }
 
+// fullColumnIDs returns the index column IDs including any implicit column IDs
+// for non-unique indexes.
+func (desc *IndexDescriptor) fullColumnIDs(tableDesc *TableDescriptor) []ColumnID {
```

Comment:
nit: slight preference for making this a method on the table descriptor. also this method should assert that this index actually belongs to the table